rust/lance-index/src/scalar/btree.rs

```diff
@@ -669,13 +674,13 @@ impl BTreeLookup {
 /// Note: this is very similar to the IVF index except we store the IVF part in a btree
 /// for faster lookup
 #[derive(Clone, Debug, DeepSizeOf)]
-pub struct BTreeIndex {
+pub struct BTreeIndex<const BATCH_SIZE: u64 = DEFAULT_BTREE_BATCH_SIZE> {
```

Comment:
What's the benefit of making `BATCH_SIZE` a compile-time constant?

Reply:
Eh, it was less that I wanted to make it a compile time constant and more that I didn't have a good way of passing the batch size to `BTreeIndex::load` (a static method).  However, I realize now I can put it in the file schema.  Let me do that real quick.